benchmarks/benchmarks/common.py

```diff
@@ -10,12 +10,243 @@
+            we can't use the automatic parameter and return value wrap/unwrap
+            from `array_api_extra.testing.lazy_xp_function`, as it comes with a
+            substantial performance overhead.
+        static_argnames : Sequence[str], optional
+            Parameter for jax.jit
+
+        Sets attributes
+        ---------------
+        backend : str
+            As the parameter (for convenience of helper functions)
+        xp : namespace
+            array namespace, potentially wrapped by array_api_compat
+        func : callable
+            function to benchmark, potentially wrapped
+        warmup : bool
+            Whether setup() should run a warmup iteration
+        """
+        self.backend = backend
+        if ":" in backend:
+            backend, device = backend.split(":")
+        else:
+            device = "cuda" if backend == "cupy" else "cpu"
+
+        with safe_import() as array_api_imports:
+            # Requires scipy >=1.16
+            from scipy._lib._array_api import array_namespace, xp_capabilities_table
+            from scipy.conftest import xp_available_backends, xp_known_backends
+        if array_api_imports.error:
+            # On older scipy versions, disregard SCIPY_ARRAY_API
+            import numpy as np
+            def array_namespace(*args, **kwargs):
+                return np
+            xp_capabilities_table = {}
+            xp_available_backends = {"numpy": np}
+            xp_known_backends = {"numpy"}
+
+        # If new backends are added to conftest.py, you need to add them here too
+        assert not xp_known_backends - set(n.split(":")[0] for n in self.backends)
+
+        try:
+            xp = xp_available_backends[backend]
+        except KeyError:
+            raise SkipNotImplemented(
+                f"{backend} not available or skipped by SCIPY_ARRAY_API")
+
+        if func and func in xp_capabilities_table:
+            capabilities = xp_capabilities_table[func]
+            skips = {n for n, _ in capabilities["skip_backends"]}
+            skips |= {n for n, _ in capabilities["xfail_backends"]}
+            if (((capabilities["cpu_only"] and device != "cpu")
+                 or (capabilities["np_only"] and backend != "numpy"))
+                and backend not in capabilities["exceptions"]):
+                skips.add(backend)
+            if backend in skips:
+                raise SkipNotImplemented(f"{backend} skipped by @xp_capabilities")
+        else:
+            capabilities = {"jax_jit": False}
+
+        # Potentially wrap namespace with array_api_compat
+        xp = array_namespace(xp.empty(0))
+
+        self.xp = xp
+        self.func = func
+        self.warmup = False
+
+        if backend == "torch":
+            import torch
+
+            torch.set_default_dtype(torch.float64)
+            try:
+                torch.empty(0, device=device)
+            except RuntimeError:
+                raise SkipNotImplemented(f"{device=} not available")
```

Comment:
This is needed for me when running in the array-api pixi env on M1 mac

```suggestion
            try:
                torch.empty(0, device=device)
            except (RuntimeError, AssertionError):
                raise SkipNotImplemented(f"{device=} not available")
```